### ISSUE TITLE
Exit upon async completion

### ISSUE THREAD

**Issue by brettz9** [YOU54F/cypress-multi-reporters#60]
### Expected behavior

Hi,

I would like to be able to have reporter behaviors run asynchronously while still being able to exit when complete.

I've added this to the end of the MultiReporters constructor (within the `execute` block) and it has been working fine:

```js
        return Promise.all(this._reporters).then(function () {
          process.exit();
        });
```

### Actual behavior

My reporters (e.g, to build mocha badges) otherwise either cannot work asynchronously or do not know when to exit (unless they coordinate with one another).

Would you be open to a PR to add this behavior?

Thanks!

**Comment by YOU54F** [comment]
Hey, sounds like a good idea, I've got a few personal and work projects that I am busy with at the moment, but will take a look, if you do open a PR I am happy to review 👍

**Comment by brettz9** [comment]
I have since discovered that:

1. There is already a Mocha-standard mechanism for asynchrous reporters (the `done` method--used already in MMR/CMR), so should probably only use that.
2. In my case, the `process.exit()` I was looking to add upon test completion could be avoided by my ensuring that the test servers I had set up had `close()` called on them.

While some might still want an option to auto-exit and force close any unclosed resources, I'm satisfied closing the issue now that I am clear that this shouldn't happen if I am closing resources successfully myself.
